src/mds/Server.cc

```diff
@@ -4452,6 +4452,26 @@ void Server::handle_client_openc(MDRequestRef& mdr)
     return;
   }
 
+  CF_MDS_RetryRequestFactory cf(mdcache, mdr, false);
+  int r = mdcache->path_traverse(mdr, cf, req->get_filepath(), 0, NULL);
+  if (r > 0) return;
+  if (r == 0) {
+    // it existed.
+    handle_client_open(mdr);
+    return;
+  }
+  if (r < 0 && r != -CEPHFS_ENOENT) {
```

Comment:
What happens if `r < 0` and r == `-CEPHFS_ENOENT`? In that case this `if` block will be skipped and will continue execution of code below this `if` block, would that lead to any error/failure?

what about
```
  if (r < 0) {
    if (r == -CEPHFS_ESTALE) {
      stmts;
    } else {
      failure stmt;
      respond_to_request();
    }
    return;
  }
```
This assumes `-CEPHFS_ENOENT` to be an error that needs to be responded back to the client, if it needs to be handled in some other way, then it's a different case.